spyderlib/plugins/shortcuts.py

```diff
@@ -7,179 +7,504 @@
+              "Pause", "Print", "Clear", "Home", "End", "Left",
               "Up", "Right", "Down", "PageUp", "PageDown"] + \
-             ["F%d" % _i for _i in range(1, 36)] + \
-             ["Space", "Exclam", "QuoteDbl", "NumberSign", "Dollar", "Percent",
-              "Ampersand", "Apostrophe", "ParenLeft", "ParenRight", "Asterisk",
-              "Plus", "Comma", "Minus", "Period", "Slash"] + \
+             ["Space", "Exclam", "QuoteDbl", "NumberSign", "Dollar",
+              "Percent", "Ampersand", "Apostrophe", "ParenLeft",
+              "ParenRight", "Asterisk", "Plus", "Comma", "Minus",
+              "Period", "Slash"] + \
              [str(_i) for _i in range(10)] + \
-             ["Colon", "Semicolon", "Less", "Equal", "Greater", "Question",
-              "At"] + [chr(_i) for _i in range(65, 91)] + \
-             ["BracketLeft", "Backslash", "BracketRight", "Underscore"]
-
-
-class Key(object):
-    MODIFIERS = {Qt.NoModifier: "", Qt.ShiftModifier: "Shift",
-                 Qt.ControlModifier: "Ctrl", Qt.AltModifier: "Alt",
-                 Qt.MetaModifier: "Meta"}
-    if sys.platform == 'darwin':
-        MODIFIERNAMES = {Qt.NoModifier: "", Qt.ShiftModifier: "Shift",
-                         Qt.ControlModifier: "Cmd", Qt.AltModifier: "Alt",
-                         Qt.MetaModifier: "Ctrl"}
-    elif sys.platform == 'win32':
-        MODIFIERNAMES = {Qt.NoModifier: "", Qt.ShiftModifier: "Shift",
-                         Qt.ControlModifier: "Ctrl", Qt.AltModifier: "Alt",
-                         Qt.MetaModifier: "Win"}
-    else:
-        MODIFIERNAMES = {Qt.NoModifier: "", Qt.ShiftModifier: "Shift",
-                         Qt.ControlModifier: "Ctrl", Qt.AltModifier: "Alt",
-                         Qt.MetaModifier: "Meta"}
-    KEYS = {}
-    for attr in KEYSTRINGS:
-        # ISSUE
-        # This is needed to cope with a strange behaviour of Qt.
-        # Qt.Key_PageUp/Qt.Key_PageDown when string parsed use PgDown/PgUp
-        attr2 = attr
-        if attr == 'PageDown':
-            attr2 = 'PgDown'
-        elif attr == 'PageUp':
-            attr2 = 'PgUp'
-
-        KEYS[getattr(Qt, "Key_"+attr)] = attr2
-
-    def __init__(self, key, mod1=Qt.NoModifier, mod2=Qt.NoModifier,
-                 mod3=Qt.NoModifier):
-        modifiers = [mod1, mod2, mod3]
-        assert all([mod in self.MODIFIERS for mod in modifiers])
-        self.modifiers = sorted(modifiers)
-        assert key in self.KEYS
-        self.key = key
+             ["Colon", "Semicolon", "Less", "Equal", "Greater",
+              "Question", "At"] + [chr(_i) for _i in range(65, 91)] + \
+             ["BracketLeft", "Backslash", "BracketRight", "Underscore",
+              "Control", "Alt", "Shift", "Meta"]
+VALID_SINGLE_KEYS = [getattr(Qt, 'Key_{0}'.format(k)) for k in SINGLE_KEYS]
+VALID_KEYS = [getattr(Qt, 'Key_{0}'.format(k)) for k in KEYSTRINGS+SINGLE_KEYS]
+VALID_ACCENT_CHARS = "ÁÉÍOÚáéíúóàèìòùÀÈÌÒÙâêîôûÂÊÎÔÛäëïöüÄËÏÖÜñÑ"

```

Comment:
This is something I ledt unfinished... but it is to be used in the search box of the... and since at some point we will have localization of the shortcut names, it made sense to include this... but probably there is a better way of doing this